dev/ProgressRing/ProgressRing.xaml

```diff
@@ -2,68 +2,100 @@
+    xmlns:local="using:Microsoft.UI.Xaml.Controls"
+    xmlns:animatedVisuals="using:AnimatedVisuals">
 
     <Style TargetType="local:ProgressRing">
         <Setter Property="Foreground" Value="{ThemeResource SystemControlHighlightAccentBrush}" />
         <Setter Property="Background" Value="{ThemeResource SystemControlBackgroundBaseLowBrush}" />
         <Setter Property="StrokeThickness" Value="4" />
+        <Setter Property="Maximum" Value="100" />
         <Setter Property="Template">
             <Setter.Value>
                 <ControlTemplate TargetType="local:ProgressRing">
                     <Grid x:Name="LayoutRoot" Background="Transparent">
-                        <!-- Gray outline of Progress Ring -->
-                        <Path
-                            Fill="Transparent"
-                            Stroke="{TemplateBinding Background}"
-                            StrokeThickness="{TemplateBinding StrokeThickness}"
-                            StrokeStartLineCap="Round"
-                            StrokeEndLineCap="Round">
-                            <Path.Data>
-                                <PathGeometry>
-                                    <PathGeometry.Figures>
-                                        <PathFigureCollection>
-                                            <PathFigure x:Name="OutlineFigurePart">
-                                                <PathFigure.Segments>
-                                                    <PathSegmentCollection>
-                                                        <ArcSegment
+
+                        <VisualStateManager.VisualStateGroups>
+                            <VisualStateGroup x:Name="CommonStates">
+                                <VisualState x:Name="Normal" />
+                                <VisualState x:Name="Determinate" />
+                                <VisualState x:Name="Indeterminate">
+                                    <Storyboard>
+                                        <DoubleAnimation
+                                            Storyboard.TargetName="LottieRoot"
+                                            Storyboard.TargetProperty="Opacity"
+                                            To="1"
+                                            Duration="0:0:0.25" />
+                                        <DoubleAnimation
+                                            Storyboard.TargetName="DeterminateProgressRingRoot"
+                                            Storyboard.TargetProperty="Opacity"
+                                            To="0"
+                                            Duration="0:0:0.25" />
+                                    </Storyboard>
+                                </VisualState>
+                            </VisualStateGroup>
+                        </VisualStateManager.VisualStateGroups>
+                        <!-- Lottie -->
+                        <Border x:Name="LottieRoot" Opacity="0">
```

Comment:
>LottieRoot [](start = 40, length = 10)

Is this border needed? can you just have the storyboard above target the player itself?

Reply:
The documentation sample has a border, so I figured it might be a standard pattern to add a border. It is not needed so I removed it.

Reply:
The Xaml framework is frequently criticized for being slow. Perhaps the largest contributor to our performance degradation is the number of elements in the tree, so we should be trying to cut elements whenever we can.

---
In reply to: [369288929](https://github.com/microsoft/microsoft-ui-xaml/pull/1858#discussion_r369288929) [](ancestors = 369288929)